### ISSUE TITLE
Memory Usage

### ISSUE THREAD

**Issue by Yanbo-23** [btsmart/splatt3r#34]
Thank you very much for your excellent work! May I ask if using eight 4090 GPUs would be sufficient to meet the memory requirements?

**Comment by btsmart** [comment]
Hello @Yanbo-23,

If you are willing to adjust the batch size, and potentially perform gradient accumulation by adjusting `accumulate_grad_batches` in the Lightning trainer object (https://lightning.ai/docs/pytorch/stable/common/trainer.html#trainer-class-api), you should be able to recreate our results on smaller hardware. I was able to complete a training run on a single 12GB RTX 2080Ti, so I don't anticipate you will have any difficulties on eight RTX 4090s.

Thanks,
Brandon

**Comment by Yanbo-23** [comment]
Thank you for your prompt response, it has been very helpful to me!

**Comment by sfchng** [comment]
Hi Brandon

I have been trying to train the model with a reduced batch size of 2 on four 48GB A6000. However the training always fail due to running out of GPU memory. Could you clarify why GPU memory usage keeps increasing as training progresses

**Comment by brandonsmartdev** [comment]
Hello @sfchng,

I believe the GPU memory usage increases during training due to the 3D Gaussian renderer. As training progresses and the 3D Gaussian primitives become larger, the number of rasterization tiles used by each Gaussian increases, which I believe increases the memory required to render them. In particular, we have noticed that if training destabilizes, the loss function encourages the network to predict very large 3D Gaussian primitives, perhaps to make up for instances where the predicted geometry is offset from the true geometry.

You can find some more information in [this issue](https://github.com/btsmart/splatt3r/issues/18). If you are having trouble training the model, you might want to consider decreasing the learning rate, or using a multi-step learning rate. If you have success with any other methods, please let us know!

Thanks,
Brandon

**Comment by saulgooodman** [comment]
Hello, I'd like to ask how long it took to train your Splat3r on a 2080 Ti?

**Comment by brandonsmartdev** [comment]
Hi @saulgooodman,

Unfortunately I no longer have the logs for the training run completed on a 2080 Ti. I remember running overnight training runs and seeing meaningful results that converged relatively quickly, but I suspect a training run that matches the metrics reported in the paper might take a few days. Beyond this I no longer have an exact answer, sorry.
